### DISCUSSION TITLE
Add environment variable support to EditableTables

### DISCUSSION THREAD

**Discussion by Mosnar** [craftcms/cms#10143] [Ideas] [answered]
In the spirit of #10111 - it would be great if we could use the autosuggested environment variables and aliases within an editable table field. There's already a "template" column type, so it should be a small change. My use-case is that I have a plugin that connects to an external API that can have multiple nodes for redundancy, and ideally a user should be able to use environment variables here.

**Comment by brandonkelly** [comment, marked as answer]
Just added this for the next release (bf8b0a6ab7e4bcb8a3d40c650934689c7073e416).

There’s a new `autosuggest` column type which supports setting `suggestEnvVars` and `suggestAliases` keys to `true` on the column definition:

```twig
{{ forms.ediatbleTableField({
    staticRows: true,
    cols: {
        secretKey: {
             type: 'autosuggest',
             heading: 'Secret key'|t('plugin-handle'),
             suggestEnvVars: true,
        },
    },
}) }}
```

Also added support for `suggestEnvVars` + `suggestAliases` to the `template` column type.

Note that for both of these column types, autosuggestions will only work for rows that were included on the initial page load. So best reserved for editable tables where `staticRows` is set to `true`.

**Reply by brandonkelly** [reply]
3.7.24 is out now with that change.
